bokeh/io.py

```diff
@@ -658,3 +664,75 @@ def export(obj, filename=None):
     image.save(filename)
 
     return os.path.abspath(filename)
+
+def _get_svgs(obj):
+    webdriver = import_required('selenium.webdriver',
+                                'To use bokeh.io.export you need selenium ' +
+                                '("conda install -c bokeh selenium" or "pip install selenium")')
+    # assert that phantomjs is in path for webdriver
+    detect_phantomjs()
+
+    html_path = tempfile.NamedTemporaryFile(suffix=".html").name
+    save(obj, filename=html_path, resources=INLINE, title="")
+
+    driver = webdriver.PhantomJS()
+    driver.get("file:///" + html_path)
+
+    _wait_until_render_complete(driver)
+
+    svg_script = """
+    var serialized_svgs = [];
+    var svgs = document.getElementsByClassName('bk-root')[0].getElementsByTagName("svg");
+    for (var i = 0; i < svgs.length; i++) {
+        var source = (new XMLSerializer()).serializeToString(svgs[i]);
+        serialized_svgs.push(source);
+    };
+    return serialized_svgs
+    """
+
+    svgs = driver.execute_script(svg_script)
+
+    driver.quit()
+
+    return svgs
+
+def export_svgs(obj, filename=None):
+    ''' Export the SVG-enabled plots within a layout. Each plot will result
+    in a distinct SVG file.
+
+    If the filename is not given, it is derived from the script name
+    (e.g. ``/foo/myplot.py`` will create ``/foo/myplot.svg``)
+
+    Args:
+        obj (LayoutDOM object) : a Layout (Row/Column), Plot or Widget object to display
+
+        filename (str, optional) : filename to save document under (default: None)
+            If None, infer from the filename.
+
+    Returns:
+        filename (str) : the filename or list of filenames where the SVGs files
+            are saved.
+
+    .. warning::
+        Responsive sizing_modes may generate layouts with unexpected size and
+        aspect ratios. It is recommended to use the default ``fixed`` sizing mode.
+
+    '''
+    svgs = _get_svgs(obj)
+
+    if len(svgs) == 0:
+        logger.warn("No SVG Plots were found.")
+        return
+
+    if filename is None:
+        filename = _detect_filename("svg")
+
+    for i, svg in enumerate(svgs):
```

Comment:
More tests are always welcome but not sure what specifically you are referring to as brittle here